modules/ethereum/src/lib.rs

```diff
@@ -659,7 +659,21 @@ pub fn verify_transaction_finalized<S: Storage>(
 	};
 	let (finalized_number, finalized_hash) = storage.finalized_block();
 
-	if header.number > finalized_number || (header.number == finalized_number && block != finalized_hash) {
+	// if header is not yet finalized => return
+	if header.number > finalized_number {
+		return false;
+	}
+
+	// check if header is actually finalized
+	let is_finalized = match header.number < finalized_number {
```

Comment:
Clippy would be mad if I didn't say anything :P
 https://rust-lang.github.io/rust-clippy/master/index.html#match_bool